packages/scene-composer/src/components/toolbars/common/ToolbarItem.tsx

```diff
@@ -103,13 +26,18 @@ export function ToolbarItem<T extends ToolbarItemOptions>({
   const itemContainerRef = useRef<HTMLDivElement | null>(null);
 
   function handleItemClick(item: T) {
+    type === 'mode-select' && setSelectedItem(item);
     onClick && onClick(item);
   }
 
   useEffect(() => {
     initialSelectedItem && setSelectedItem(initialSelectedItem);
   }, [initialSelectedItem]);
```

Comment:
how come the initial selected item overrides the selected item when it changes after the initial render?

Reply:
I'm not sure exactly.. that's some legacy code that need some refactoring... but I don't want to do too much refactoring in my feature change since it will make the merge hard.